app/assets/javascripts/service_types/IsServiceWorking.js

```diff
@@ -0,0 +1,186 @@
+        </p>
+        <p style={{padding: '10px 0 10px 10px'}}>
+          Use the buttons below to explore how students may have responded to
+          services assigned to them over the past school year.
+        </p>
+        {this.renderButtons()}
+        {this.renderCharts()}
+      </div>
+    );
+  }
+
+  renderButtons() {
+    const attendanceOfficer = {
+      id: 502,
+      name: 'Attendance Officer',
+    };
+
+    const attendanceContract = {
+      id: 503,
+      name: 'Attendance Contract',
+    };
+
+    return (
+      <div style={style.buttonSectionStyle}>
+        <Button style={style.buttonStyle}
+          onClick={() => {
+            this.setState({serviceTypeSelected: null}, () => {
+              this.setState({serviceTypeSelected: attendanceOfficer});
+            });
+          }}>
+          Attendance Officer
+        </Button>
+        <Button style={style.buttonStyle}
+          onClick={() => {
+            this.setState({serviceTypeSelected: null}, () => {
+              this.setState({serviceTypeSelected: attendanceContract});
+            });
+          }}>
+          Attendance Contract
+        </Button>
+      </div>
+    );
+  }
+
+  renderCharts() {
+    const {serviceTypeSelected} = this.state;
+
+    if (!serviceTypeSelected) return null;
+
+    return (
+      <GenericLoader
+        promiseFn={this.fetchIsServiceWorkingData}
+        render={this.renderPage} />
+    );
+  }
+
+  renderPage(json) {
+    return (
+      <div style={style.container}>
+        {this.renderStudents(json)}
+      </div>
+    );
+  }
+
+  renderPhasebands(services) {
+    return services.map(service => {
+      return {
+        momentUTCstart: moment.utc(service.date_started),
+        momentUTCend: moment.utc(service.discontinued_at),
+        text: this.state.serviceTypeSelected.name
+      };
+    }, this);
+  }
+
+  renderStudents(json) {
+    const chartData = json.chart_data;
+    const sortedData = _.sortBy(chartData, datum => {
+      return (datum.services.length > 0)
+        ? _.map(datum.services, 'date_started').sort()[0]
+        : 0;
+    });
+
+    return sortedData.map((datum) => {
+      const student = datum.student;
+      const services = datum.services;
+      const phasebands = this.renderPhasebands(services);
+
+      return (
+        <div key={student.id} style={style.studentRow}>
+          <a href={Routes.studentProfile(student.id)} style={style.a}>
+            {student.first_name} {student.last_name}
+          </a>
+          <span style={{marginLeft: 5}}>({datum.school} Grade {student.grade})</span>
+          <ProfileBarChart
+            events={datum.absences}
+            titleText="Absences"
+            monthsBack={24}
+            phasebands={phasebands}
+            hideBackToTop={true}
+            styles={style.profileBarChartStyles} />
+        </div>
+      );
+    }, this);
+
+  }
+
+}
+
+IsServiceWorking.propTypes = {
+  serializedData: React.PropTypes.object.isRequired,
```

Comment:
I don't think this is used, looks like `GenericLoader` is loading the data for this component.

Reply:
👍